geode-http-service/src/main/java/org/apache/geode/internal/cache/InternalHttpService.java

```diff
@@ -181,6 +181,9 @@ public synchronized void addWebApplication(String webAppContext, Path warFilePat
     webapp.setContextPath(webAppContext);
     webapp.setWar(warFilePath.toString());
     webapp.setParentLoaderPriority(false);
+
+    // GEODE-7334: do not load jodaModule even if it's present in the system classloader
+    webapp.getServerClasspathPattern().add("com.fasterxml.jackson.datatype.joda.JodaModule");
```

Comment:
I think this pattern restriction should be made more general to cover all of `com.fasterxml.jackson` as well as all of `org.springframework`. It's not just `JodaModule` that would cause this problem. Have a look at this: https://github.com/spring-projects/spring-framework/blob/809009e6c4d8653ea18e8fe9ae590242871711c4/spring-web/src/main/java/org/springframework/http/converter/json/Jackson2ObjectMapperBuilder.java#L772-L822.

The general approach should be to have the webapp be self-contained as much as possible. These classpath patterns should probably only exist as failsafes.